api/src/modules/sections/sections.module.ts

```diff
@@ -4,10 +4,16 @@ import { SectionsService } from '@api/modules/sections/sections.service';
 import { Section } from '@shared/dto/sections/section.entity';
 import { TypeOrmModule } from '@nestjs/typeorm';
 import { AuthModule } from '@api/modules/auth/auth.module';
+import { PostgresSurveyDataRepository } from '@api/infrastructure/postgres-survey-data.repository';
+import { SQLAdapter } from '@api/infrastructure/sql-adapter';
 
 @Module({
   imports: [TypeOrmModule.forFeature([Section]), forwardRef(() => AuthModule)],
-  providers: [SectionsService],
+  providers: [
+    SQLAdapter,
```

Comment:
SQLAdapter is being injected to multiple services, it should be part of/exported from some module, and then import that module where needed. This approach adds a new graph of dependencies